### PR TITLE
fix: update site address of antd 4.x

### PR DESCRIPTION
**Pull request by MuxinFeng**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close https://github.com/ant-design/ant-design/issues/41512

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    add site address of antd 5.x      |
| 🇨🇳 Chinese |   添加 antd 5.x 的文档地址        |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: site/theme/template/Layout/Header.jsx [hunk 1]
```diff
@@ -114,7 +114,6 @@ class Header extends React.Component {
       intl: { locale },
     } = this.props;
     const docVersions = {
-      ...themeConfig.docNewVersions,
```
Comment (MadCcc) [inline site/theme/template/Layout/Header.jsx:117]: 4.x 和 5.x 还是放在 3.x 上面好

Comment (MuxinFeng) [reply site/theme/template/Layout/Header.jsx:117]: 好嘞，4.x 的文档也有这个问题，下次有机会可以顺手修掉。

Comment (MuxinFeng) [reply site/theme/template/Layout/Header.jsx]: 验了下，没问题啦 
image

File: site/bisheng.config.js [hunk 2]
```diff
@@ -74,8 +74,9 @@ module.exports = {
       '0.9.x': 'http://09x.ant.design',
     },
     docNewVersions: {
-      '4.x': 'https://4x.ant.design',
-    },
+      '5.x': 'https://ant.design',
```
[the remark this answers is not in the document]
Comment (MuxinFeng) [reply site/bisheng.config.js]: 换了种一劳永逸的写法，大佬再给看看吧~

**Comment by MuxinFeng** [comment]
我先给 5.x 提个？可以看 preview 效果

**Comment by MadCcc** [comment]
4.x 的链接还是可以改下吧

File: site/bisheng.config.js [hunk 3]
```diff
@@ -74,8 +74,8 @@ module.exports = {
       '0.9.x': 'http://09x.ant.design',
     },
     docNewVersions: {
-      '4.x': 'https://4x.ant.design',
-    },
+      '4.x': 'http://4x.ant.design',
```
Comment (yoyo837) [inline site/bisheng.config.js:77]: 这是什么操作?

Comment (MuxinFeng) [reply site/bisheng.config.js:77]: 一开始参考 antd 4.x 那边的文档，发现给老版本的都是 http，然后猜测站点会根据 url 严格匹配（包括协议），然后就照着改了下，最后发现开发环境下能跳转成功，就推上来了。我再去验证下哈。

Comment (MadCcc) [reply site/bisheng.config.js:77]: 这个也是有效的，3.x 网站发一下就好了，这里可以不用改了

Comment (MuxinFeng) [reply site/bisheng.config.js:77]: 好滴，那我先关掉啦~

**Pull request closed without merging**